### PR TITLE
:recycle: Refactor `daily_material` sort_item

### PR DESCRIPTION
**Pull request by xr1s**
## 描述 / Description

优化排序逻辑代码，不影响排序输出。

### CODE REVIEW

**Comment by StellaCaerulea** [comment]
比以前好太多。之前那玩意谁写的？笑烂

**Comment by luoshuijs** [comment]
@StellaCaerulea 你直接去看 commit 历史啊.jpg

File: plugins/genshin/daily/material.py [hunk 1]
```diff
@@ -57,27 +57,20 @@ def sort_item(items: List["ItemData"]) -> Iterable["ItemData"]:
...
-                    lambda y: (
-                        ArkoWrapper(y[1])
-                        .sort(lambda z: z.refinement or z.constellation or -1, reverse=True)
-                        .groupby(lambda z: z.refinement or z.constellation or -1)  # 根据命座/精炼进行分组并排序
-                        .map(lambda i: ArkoWrapper(i[1]).sort(lambda j: j.id))
-                    )
-                )
-            )
+
+    def key(item: ItemData):
```
Comment (omg-xtao) [inline plugins/genshin/daily/material.py]:
```suggestion
    def key(item: "ItemData"):
```

**Review by omg-xtao — CHANGES_REQUESTED** [review]

File: plugins/genshin/daily/material.py (hunk 1 above)
Comment (StellaCaerulea) [reply plugins/genshin/daily/material.py]: Why this?

Comment (omg-xtao) [reply plugins/genshin/daily/material.py]: Using variable 'ItemData' before assignment

Comment (Billyzou0741326) [reply plugins/genshin/daily/material.py]: > Why this?

https://github.com/PaiGramTeam/PaiGram/blob/3ac0bdfec672f1691acedb884acd493532d936bf/plugins/genshin/daily/material.py#L55

Comment (luoshuijs) [reply plugins/genshin/daily/material.py]: Are you looking for deferred type hint？

**Review by omg-xtao — APPROVED** [review]

**Review by luoshuijs — APPROVED** [review]

**Review by Billyzou0741326 — APPROVED** [review]

**Review by StellaCaerulea — APPROVED** [review]
善

**Pull request merged**